model/job-management/src/main/proto/beam_artifact_api.proto

```diff
@@ -31,8 +31,77 @@ option java_outer_classname = "ArtifactApi";
 
 import "beam_runner_api.proto";
 
-// A service to stage artifacts for use in a Job.
+// A service to retrieve artifacts for use in a Job.
+service ArtifactRetrievalService {
+  // Resolves the given artifact reference into one or more simpler artifact
+  // references (e.g. a Maven dependency into a (transitive) set of jars.
+  // If no further simplification is possible, returns the original artifacts,
+  // at which point, all artifacts must be gettable.
+  rpc ResolveArtifact(ResolveArtifactRequest) returns (ResolveArtifactResponse);
+
+  // Retrieves the given artifact as a stream of bytes.
+  rpc GetArtifact(GetArtifactRequest) returns (stream GetArtifactResponse);
+}
+
+// A service that allows the client to act as an ArtifactRetrievalService,
+// for a particular job with the server initiating requests and receiving
+// responses.
+// A client calls the service with an ArtifactResponseWrapper that has the
+// staging token set, and thereafter responds to the server's requests.
 service ArtifactStagingService {
```

Comment:
I can see that your trying to set this up so that the staging service can make calls to the SDK so that this way it can get artifacts that are created by the expansion service.